validation_testing/scripts/glue_job_synchronous_execution.py

```diff
@@ -0,0 +1,35 @@
+import boto3
+import sys
+import time
+
+glue = boto3.client('glue')
+poll_interval = 10  # seconds
+max_polls = 120 # allow 20 minutes (10 * 120 = 1200 seconds = 1200/60 = 20 minutes)
+
+
+def start_job_and_wait_for_completion(job_name):
+    start_job_response = glue.start_job_run(JobName=job_name)
+    job_run_id = start_job_response['JobRunId']
+    for polls in range(max_polls):
+        status = glue.get_job_run(JobName=job_name, RunId=job_run_id)['JobRun']['JobRunState']
+        if status == 'SUCCEEDED':
+            break 
+        elif status in ['FAILED', 'STOPPED', 'TIMEOUT']:
+            print('Job failed')
+            break
+        elif polls >= max_polls:
+            print('Exceeded max polls')
```

Comment:
would we have to manually check the job if it timed out? if so maybe we should add a warning in the logs.

Reply:
Addressing both of your comments at once - if there is an issue with the glue job, either bc it didn't finish within our 20 min polling limit, or the job itself timed out, or the job itself failed or stopped, we will fail the full release tests. Then in the logs we would see that `echo "FINISHED RUNNING GLUE JOBS FOR ${CONNECTOR_NAME}."` didn't run and the last echo'd line was "Starting glue jobs...", so we would already know the issue was with the glue job.

In terms of next steps, yes we would then have to go check the glue job itself. But I don't think that's a problem, we will need to go investigate in the console regardless of if we knew it was a timeout or failure. And the reason for fast-failing here, instead of allowing the infra destruction to happen, is so we don't have to deal with the glue job being torn down before we can investigate.

So that was my reasoning behind this set up, let me know if you disagree with any part of it